vyper/semantics/analysis/base.py

```diff
@@ -97,6 +97,14 @@ def from_abi(cls, abi_dict: Dict) -> "StateMutability":
         #       specifying a state mutability modifier at all. Do the same here.
 
 
+class Modifiability(enum.IntEnum):
+    MODIFIABLE = enum.auto()
+    IMMUTABLE = enum.auto()
+    NOT_MODIFIABLE = enum.auto()
```

Comment:
what is the difference between immutable and not modifiable?

Reply:
i see NOT_MODIFIABLE is applied to calldata args, maybe it's the same as/can be merged with CONSTANT_IN_CURRENT_TX. or maybe we can just rename NOT_MODIFIABLE to CALLDATA!

Reply:
Immutables are modifiable in ctor but not any other function, so it could be useful if we need to distinguish it in the ctor, although that is currently already [handled](https://github.com/vyperlang/vyper/blob/bf26f83ec21788d9fb591879fdd6af2ddfacb050/vyper/codegen/expr.py#L191).

I think renaming NOT_MODIFIABLE to CALLDATA sounds good. It could be useful to preserve the granularity in the event it helps with some refactoring.